tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,241 @@
+  attributes(d)=a
+  h5write(obj=d,file=nf,
+          name=paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]),
+          write.attributes=T)
+}
+```
+
+## Spectral Subsetting
+
+The goal of subsetting this dataset is to substantially reduce the file size to make it faster to download and process these data. While each AOP mosaic tile is not particularly large in terms of its spatial scale (1km by 1km at 1m resolution= 1,000,000 pixels, or about half as many pixels at shown on a standard 1080p computer screen), the 426 spectral bands available result in a fairly large file size. While having hundreds of bands is important for advanced research, we can use far fewer bands for demonstartion and teaching purposes, while still getting a good sense of what these hyperspectral data can do. Therefore, we will reduce the spectral resolution of these data by selecting every fourth band in the dataset, which reduces the file size to 1/4 of the original!
+
+```{r subset-wavelengths}
+#First, we make out 'index'  a list of number that will allow us to select every fourth band
+idx <- seq(from = 1, to = 426, by = 4)
+
+# We then use this index to select particular wavelengths from the full tile using the "index=" argument
+ws <- h5read(file = f, 
+             name = "SJER/Reflectance/Metadata/Spectral_Data/Wavelength", 
+             index = list(idx)
+            )
+
+# As per above, we also need the wavelength attributes
+wa <- h5readAttributes(file = f, 
+                       name = "SJER/Reflectance/Metadata/Spectral_Data/Wavelength")
+attributes(ws) <- wa
+
+# Finally, write the subset of wavelengths and their attributes to the subset file
+h5write(obj=ws, file=nf,
+        name="SJER/Reflectance/Metadata/Spectral_Data/Wavelength",
+        write.attributes=T)
+```
+
+## Spatial Subsetting
+
+Even after spectral subsetting, our file size would be greater than 100Mb. Therefore, we will perform a spatial subsetting process as well when we read the hyperspectral imagery data. Note that the `index=` argument in `h5read` requires a `list()` object of three dimenstions for this example - in the order of band, column, row. Here, we are taking every fourth band (using our `idx` variabe), columns 501:1000 (the eastern half of the tile) and rows 1:500 (the northern half of the tile). The results in us extracting every fourth band of the northeast quadrant of the mosaic tile.
+
+
+```{r read-hyp-data}
+# Read in reflectance data - note the index argument! This is ultimately how we reduce the file size.
+hs <- h5read(file = f, 
+             name = "SJER/Reflectance/Reflectance_Data", 
+             index = list(idx, 501:1000, 1:500)
+            )
+# grab the '$dim' attribute - as this will be needed when writing the file at the bottom
+hsd=attributes(hs)
+
+# We also need the attributes for the reflectance data, of course.
+ha <- h5readAttributes(file = f, 
+                       name = "SJER/Reflectance/Reflectance_Data")
+
```

Comment:
Seems like this could use a more full explanation in text not just in comments. And then the code in a new code chunk.

Reply:
Good point - I have added a paragraph and separated this part of the code into two chunks.